httpserver/swarm.cpp

```diff
@@ -149,16 +166,18 @@ swarm_id_t get_swarm_by_pk(const std::vector<SwarmInfo>& all_swarms,
             cur_best = si.swarm_id;
             cur_min = dist;
         }
+
+        /// Find the letfmost
+        if (si.swarm_id < leftmost_id) {
+            leftmost_id = si.swarm_id;
+        }
     }
 
     // handle special case
-
-    if (res > all_swarms[0].swarm_id) {
-        uint64_t dist =
-            std::numeric_limits<uint64_t>::max() - res + all_swarms[0].swarm_id;
-
+    if (res > leftmost_id) {
```

Comment:
As discussed offline, I there are 2 potential issues here
You need to do the wrapping check for *both* the left and rightmost
And the `res > leftmost_id` check is not taking into consideration the case where the leftmost swarm id is on the right of the middle value

Reply:
Updated the commit to check wrapping to the rightmost id as well. As for the case where leftmost is on the right half of possible values, we don't need to account for that since the shortest distance to that id would be the direct distance between two values, without wrapping from the right side. The same logic applies to the rightmost id.

Reply:
The logic seems good, although I think the comparison could be more explicit, if I understand it correctly. This is about the "wrapped distance" right? If so, we're only interested to check if we're below or above leftmost_id and rightmost_id respectively:
```cpp
if (res > rightmost_id) {
  ...
} else if (res < leftmost_id) {
   ...
}
```
This will produce the same result but would probably be more readable.

Reply:
Also for posterity, the reason we *don't* need to check on the other side of the left or rightmost id is because even if the leftmost id is on the right half of the range, either 1) there is another id to the right of it (i.e. this other id would be closer when wrapping to the right), or 2) the leftmost id is the only id in which case it would necessarily also be the *rightmost* id and we would check wrapping from the right anyway

Reply:
The reason I use `res > leftmost_id` is to make it clear that `(MAX_ID - res) + leftmost_id;` will not overflow.

Reply:
Modified to use Sacha's suggestion.